lib/internal/timers.js

```diff
@@ -520,16 +520,25 @@ function getTimerCallbacks(runNextTicks) {
 
     let ranAtLeastOneTimer = false;
     let timer;
+    const secondaryList = timerListQueue.secondary();
     while ((timer = L.peek(list)) != null) {
       const diff = now - timer._idleStart;
 
       // Check if this loop iteration is too early for the next timer.
       // This happens if there are more timers scheduled for later in the list.
-      if (diff < msecs) {
+      //
+      // Else, if the secondary list is earlier than current timer, then we
+      // return to the main loop to process the secondary list first.
+      if (diff < msecs || secondaryList?.expiry < timer._idleStart + msecs) {
```

Comment:
Is there any significant change in the benchmark if you put `secondaryList?.expiry` inside a const variable instead of using `seconaryList`?